sdks/python/apache_beam/examples/game_stats_it_test.py

```diff
@@ -0,0 +1,153 @@
+    _unique_id = str(uuid.uuid4())
+
+    # Set up PubSub environment.
+    from google.cloud import pubsub
+    self.pubsub_client = pubsub.Client(project=self.project)
+    unique_topic_name = self.INPUT_TOPIC + _unique_id
+    unique_subscrition_name = self.INPUT_SUB + _unique_id
+    self.input_topic = self.pubsub_client.topic(unique_topic_name)
+    self.input_sub = self.input_topic.subscription(unique_subscrition_name)
+
+    self.input_topic.create()
+    test_utils.wait_for_topics_created([self.input_topic])
+    self.input_sub.create()
+
+    # Set up BigQuery environment
+    from google.cloud import bigquery
+    client = bigquery.Client()
+    unique_dataset_name = self.OUTPUT_DATASET + str(int(time.time()))
+    self.dataset = client.dataset(unique_dataset_name, project=self.project)
+    self.dataset.create()
+
+    self._test_timestamp = int(time.time() * 1000)
+
+  def _inject_pubsub_game_events(self, topic, message_count):
+    """Inject game events as test data to PubSub."""
+
+    logging.debug('Injecting %d game events to topic %s',
+                  message_count, topic.full_name)
+
+    for _ in range(message_count):
+      topic.publish(self.INPUT_EVENT % self._test_timestamp)
+
+  def _cleanup_pubsub(self):
+    test_utils.cleanup_subscriptions([self.input_sub])
+    test_utils.cleanup_topics([self.input_topic])
+
+  @attr('GS')
+  def test_game_stats_it(self):
+    state_verifier = PipelineStateMatcher(PipelineState.RUNNING)
+
+    success_condition = 'mean_duration=300 LIMIT 1'
+    sessions_query = ('SELECT mean_duration FROM [%s:%s.%s] '
+                      'WHERE %s' % (self.project,
+                                    self.dataset.name,
+                                    self.OUTPUT_TABLE_SESSIONS,
+                                    success_condition))
+    bq_sessions_verifier = BigqueryMatcher(self.project,
+                                           sessions_query,
+                                           self.DEFAULT_EXPECTED_CHECKSUM)
+
+    # TODO(mariagh): Add teams table verifier once game_stats.py is fixed.
+
+    extra_opts = {'subscription': self.input_sub.full_name,
+                  'dataset': self.dataset.name,
+                  'topic': self.input_topic.full_name,
+                  'fixed_window_duration': 1,
+                  'user_activity_window_duration': 1,
+                  'wait_until_finish_duration':
+                      self.WAIT_UNTIL_FINISH_DURATION,
+                  'on_success_matcher': all_of(state_verifier,
+                                               bq_sessions_verifier)}
+
+    # Register cleanup before pipeline execution.
+    self.addCleanup(self._cleanup_pubsub)
+    self.addCleanup(utils.delete_bq_table, self.project,
+                    self.dataset.name, self.OUTPUT_TABLE_SESSIONS)
+    self.addCleanup(utils.delete_bq_table, self.project,
+                    self.dataset.name, self.OUTPUT_TABLE_TEAMS)
+
```

Comment:
Please also cleanup the dataset that was create at setup time.

Reply:
Will do.

Reply:
I have added cleanups for datasets in other two tests.